File: BAEKJOON/easy/10845.py
import sys
sys.stdin = open("input.txt", "r")


n=int(input())
queue=[]
for _ in range(n):
    command=input().strip().split() # split()을 안하면, push 1이 하나의 문자열로 들어온다. strip()은 공백제거
    cmd=command[0]
    if cmd=='push':
        queue.append(command[1])
    elif cmd=='pop':
        if queue:
            print(queue.pop(0))
        else:
            print(-1)
    elif cmd=='size':
        print(len(queue))
    elif cmd=='empty':
        if queue:
            print(0)
        else:
            print(1)
    elif cmd=='front':
        if queue:
            print(queue[0])
        else:
            print(-1)
    elif cmd=='back':
        if queue:
            print(queue[-1])
        else:
            print(-1)

File: BAEKJOON/easy/10845.cpp
#include <iostream>

using namespace std;

int main(){
  int n;
  cin >> n;
  int queue[10000];
  int front = 0, back = 0;
  for(int i = 0; i < n; i++){
    string command;
    cin >> command;
    if(command == "push"){
      int x;
      cin >> x;
      queue[back++] = x;
    } else if(command == "pop"){
      if(front == back) cout << -1 << '\n';
      else cout << queue[front++] << '\n';
    } else if(command == "size"){
      cout << back - front << '\n';
    } else if(command == "empty"){
      if(front == back) cout << 1 << '\n';
      else cout << 0 << '\n';
    } else if(command == "front"){
      if(front == back) cout << -1 << '\n';
      else cout << queue[front] << '\n';
    } else if(command == "back"){
      if(front == back) cout << -1 << '\n';
      else cout << queue[back - 1] << '\n';
    }
  }
  

}